python/hail/expr/expression.py

```diff
@@ -885,9 +901,7 @@ def product(self):
         :class:`.NumericExpression`
             The product of the collection.
         """
-        return self._method("product",
-                            TInt64() if isinstance(self._type.element_type, TInt32) or
-                                        isinstance(self._type.element_type, TInt64) else TFloat64())
+        return self._method("product", self._type.element_type)
```

Comment:
Does the type conversion get handled somewhere else now?

Reply:
this was just wrong. product returns an array of the same type.